### PR TITLE
refactor!: clean up `CoapMessage` class

### PR DESCRIPTION
**Pull request by JKRhb**
I noticed some more clean up potential in the `CoapMessage` class, due to the fact that you do not need to return the current object for chaining method calls because of Dart's cascade notation. I also managed to get rid of a couple of null check operators, making the package a bit more reliable.

Technically, this PR introduces small but breaking API changes, so I guess it should be deferred to a point when more critical API changes have accumulated for a new major release.

CC @JosefWN 

Depends on #89.

### CODE REVIEW

File: lib/src/coap_message.dart [hunk 1]
```diff
@@ -148,8 +146,10 @@ class CoapMessage {
   typed.Uint8Buffer? get token => _token;
 
   /// As a string
-  String get tokenString =>
-      _token != null ? CoapByteArrayUtil.toHexString(_token!) : '';
+  String get tokenString {
+    final token = _token;
```
Comment (JosefWN) [inline lib/src/coap_message.dart:150]: What is the purpose of this variable?

File: lib/src/coap_message.dart [hunk 2]
```diff
@@ -292,9 +292,10 @@ class CoapMessage {
 
   /// The payload of this CoAP message in string representation.
   String? get payloadString {
-    if (payload != null && payload!.isNotEmpty) {
+    final payload = this.payload;
```
Comment (JosefWN) [inline lib/src/coap_message.dart:295]: Also here, not entirely clear on this one.

File: lib/src/coap_message.dart (hunk 1 above)
Comment (JKRhb) [reply lib/src/coap_message.dart:150]: It is simply introduced for type promotion purposes, making it possible to get rid of the null check operator. This is not a must, as `_token` is obviously not `null` when the operator is used, but it is one potential source for bugs less in the future :)

File: lib/src/coap_message.dart (hunk 2 above)
Comment (JKRhb) [reply lib/src/coap_message.dart:295]: Same reason as above, the linter does not complain anymore when the null check is removed a couple of lines below.

File: lib/src/coap_message.dart (hunk 1 above)
Comment (JosefWN) [reply lib/src/coap_message.dart:150]: Ah, gotcha.

**Comment by JosefWN** [comment]
Yeah I actually thought about doing the same thing, good thinking 👍

Chaining isn't that critical in Dart anyway since you can chain either way with `..`. LGTM.

**Comment by JKRhb** [comment]
I now rebased this PR against #89, so that the two PRs can be merged subsequently.

**Pull request merged**